### ISSUE TITLE
[MJAVADOC-543] Archiving to jar is very slow

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#828]
**[Claes Nordmark](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nordmcla)** opened **[MJAVADOC-543](https://issues.apache.org/jira/browse/MJAVADOC-543?redirect=false)** and commented

Doing a stackdump while "maven-javadoc-plugin:3.0.1:jar (attach-javadoc)" shows thread spending a lot of time in UnixNativeDispatcher.getgrgid.

This was not visible in java 8 and javadoc plugin version 2.4.

Our previous build time has been increased from 5min to 3h 30min.

Tested version Java 11 and 3.0.1 of maven-javadoc-plugin

maven config:

 

```java
<plugin>
<groupId>org.apache.maven.plugins</groupId>
<artifactId>maven-javadoc-plugin</artifactId>
<version>3.0.1</version>
<executions>
  <execution>
    <id>attach-javadocs</id>
    <goals>
      <goal>jar</goal>
    </goals>
    <configuration>
      <failOnError>false</failOnError>
    </configuration>
    </execution>
  </executions>
</plugin>
```

 

`java.lang.Thread.State: RUNNABLE`

```
at sun.nio.fs.UnixNativeDispatcher.getgrgid(java.base@11/Native Method)  
at sun.nio.fs.UnixUserPrincipals.fromGid(java.base@11/UnixUserPrincipals.java:129)  
at sun.nio.fs.UnixFileAttributes.group(java.base@11/UnixFileAttributes.java:208)  
- locked <0x00000006a2fe0f40> (a sun.nio.fs.UnixFileAttributes)  
at sun.nio.fs.UnixFileAttributeViews$Posix.addRequestedPosixAttributes(java.base@11/UnixFileAttributeViews.java:214)  
at sun.nio.fs.UnixFileAttributeViews$Unix.readAttributes(java.base@11/UnixFileAttributeViews.java:362)  
at sun.nio.fs.AbstractFileSystemProvider.readAttributes(java.base@11/AbstractFileSystemProvider.java:94)  
at java.nio.file.Files.readAttributes(java.base@11/Files.java:1996)  
at org.codehaus.plexus.components.io.attributes.FileAttributes.<init>(FileAttributes.java:67)  
at org.codehaus.plexus.components.io.resources.PlexusIoFileResourceCollection.addResources(PlexusIoFileResourceCollection.java:156)  
at org.codehaus.plexus.components.io.resources.PlexusIoFileResourceCollection.getResources(PlexusIoFileResourceCollection.java:250)  
at org.codehaus.plexus.archiver.AbstractArchiver$1.hasNext(AbstractArchiver.java:506)  
at org.codehaus.plexus.archiver.zip.AbstractZipArchiver.createArchiveMain(AbstractZipArchiver.java:257)  
at org.codehaus.plexus.archiver.zip.AbstractZipArchiver.execute(AbstractZipArchiver.java:235)  
at org.codehaus.plexus.archiver.AbstractArchiver.createArchive(AbstractArchiver.java:987)  
at org.apache.maven.archiver.MavenArchiver.createArchive(MavenArchiver.java:649)  
at org.apache.maven.plugins.javadoc.JavadocJar.generateArchive(JavadocJar.java:300)  
at org.apache.maven.plugins.javadoc.JavadocJar.doExecute(JavadocJar.java:205)  
at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute(AbstractJavadocMojo.java:1912)  
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)  
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)  
at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:200)  
at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:196)  
at java.util.concurrent.FutureTask.run(java.base@11/FutureTask.java:264)  
at java.util.concurrent.Executors$RunnableAdapter.call(java.base@11/Executors.java:515)  
at java.util.concurrent.FutureTask.run(java.base@11/FutureTask.java:264)  
at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11/ThreadPoolExecutor.java:1128)  
at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11/ThreadPoolExecutor.java:628)  
at java.lang.Thread.run(java.base@11/Thread.java:834)  
```

---

**Affects:** 3.0.1

**Remote Links:**
- [#17 Plexus IO
](https://github.com/codehaus-plexus/plexus-io/pull/17)

**Comment by jira-importer** [comment]
**[Claes Nordmark](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nordmcla)** commented

Narrowed down to https://github.com/codehaus-plexus/plexus-io/pull/17 :) made a push for a fix to plexus-io

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [aceadd711651833de882f82eee7d55fb1db99af0](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=aceadd711651833de882f82eee7d55fb1db99af0)
